File: golang/client/client_test.go
package client

import (
	"testing"

	"github.com/alibabacloud-go/tea/tea"
	"github.com/alibabacloud-go/tea/utils"
)

func Test_GetTimestamp(t *testing.T) {
	utils.AssertEqual(t, 13, len(tea.StringValue(GetTimestamp())))
}

func Test_HasError(t *testing.T) {
	res := map[string]interface{}{
		"error_response": "ok",
	}
	out := tea.BoolValue(HasError(res))
	utils.AssertEqual(t, true, out)

	delete(res, "error_response")
	out = tea.BoolValue(HasError(res))
	utils.AssertEqual(t, false, out)

	res["errcode"] = 0
	out = tea.BoolValue(HasError(res))
	utils.AssertEqual(t, false, out)

	res["errcode"] = 10
	out = tea.BoolValue(HasError(res))
	utils.AssertEqual(t, true, out)
}

func Test_ComputeSignature(t *testing.T) {
	res := ComputeSignature(tea.String("accesskeysecret"), tea.String("canonicalString"))
	utils.AssertEqual(t, "NK+EVoCsJHmdwTSe5lzCj5GavWyQBO12rQaUfRScfz0=", tea.StringValue(res))
}

func Test_GetCanonicalStringForIsv(t *testing.T) {
	res := GetCanonicalStringForIsv(tea.String("timestamp"), tea.String("suiteTicket"))
	utils.AssertEqual(t, "timestamp\nsuiteTicket", tea.StringValue(res))

	res = GetCanonicalStringForIsv(tea.String("timestamp"), nil)
	utils.AssertEqual(t, "timestamp", tea.StringValue(res))
}

File: ts/test/client.test.ts
import Client from "../src/client";

import * as $tea from "@alicloud/tea-typescript";
import assert from "assert";
import "mocha";

describe("ding-util", function () {
  it("Module should ok", function () {
    assert.ok(Client);
  });
});

describe("getTimestamp", function () {
  it("timestamp should ok", function () {
    const timestamp = Client.getTimestamp();
    assert.equal(typeof timestamp, "string");
    assert.equal(timestamp.length, 13);
  });
});

describe("testHasError", function () {
  it("test null", () => {
    const data = null;
    const res = Client.hasError(data);
    assert.equal(res, true);
  });

  it("test undefined", () => {
    const data = undefined;
    const res = Client.hasError(data);
    assert.equal(res, true);
  });

  it("test error_response", () => {
    const data = { error_response: true };
    const res = Client.hasError(data);
    assert.equal(res, true);
  });

  it("test errcode null", () => {
    const data = { errcode: null };
    const res = Client.hasError(data);
    assert.equal(res, false);
  });

  it("test errcode undefined", () => {
    const data = { errcode: undefined };
    const res = Client.hasError(data);
    assert.equal(res, false);
  });

  it("test errcode 0", () => {
    const data = { errcode: 0 };
    const res = Client.hasError(data);
    assert.equal(res, false);
  });

  it("test errcode 500", () => {
    const data = { errcode: 500 };
    const res = Client.hasError(data);
    assert.equal(res, true);
  });
});

describe("testComputeSignature", function () {
  it("computeSignature", () => {
    assert.equal(
      Client.computeSignature("accessSecret", "test"),
      "Ku8nmti/NjJ5kJzEyzXkpgNSuzwn4TAh3rxsBSaA/1E="
    );
  });
});

describe("getCanonicalStringForIsv", function () {
  it("getCanonicalStringForIsv", () => {
    const timestamp = Client.getTimestamp();
    assert.equal(
      Client.getCanonicalStringForIsv(Client.getTimestamp(), null),
      timestamp
    );
    assert.equal(
      Client.getCanonicalStringForIsv(Client.getTimestamp(), "suiteTicket"),
      `${timestamp}\nsuiteTicket`
    );
  });
});
